xds/internal/client/load/store.go

```diff
@@ -24,8 +24,58 @@ import (
 
 const negativeOneUInt64 = ^uint64(0)
 
-// Store is a repository for LB policy implementations to report store load
-// data. It is safe for concurrent use.
+// A pair of cluster and service name. The same cluster can be used by multiple
+// services, and one service can use multiple clusters. So we need a pair with
+// both name to accurately indicate where the load belongs.
+type clusterAndServiceNames struct {
+	cluster string
+	service string
+}
+
+type Store struct {
+	clusters sync.Map // map[clusterAndServiceNames]*rpcCountData
+}
+
+// PerCluster returns the PerClusterStore for the given clusterName +
+// serviceName.
+func (ls *Store) PerCluster(clusterName, serviceName string) *PerClusterStore {
+	if ls == nil {
+		return nil
+	}
+
+	k := clusterAndServiceNames{
+		cluster: clusterName,
+		service: serviceName,
+	}
+	p, ok := ls.clusters.Load(k)
+	if !ok {
+		s := &PerClusterStore{}
+		p, _ = ls.clusters.LoadOrStore(k, s)
+	}
+	return p.(*PerClusterStore)
+}
+
+// CallServerLoad adds one server load record for the given locality. The
+// load type is specified by desc, and its value by val.
+func (ls *Store) CallServerLoad(clusterName, serviceName, locality, name string, d float64) {
```

Comment:
Why does the top level `Store` provide this method alone?

Reply:
Deleted.

Though I think we should provide all the methods to record loads, instead of a `PerCluster` method.
That will result in a big change in the load reporting client when retrieving loads. I will do it in the next PR.